pkg/gits/credentialhelper/git_credential.go

```diff
@@ -0,0 +1,122 @@
+package credentialhelper
+
+import (
+	"fmt"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+)
+
+// GitCredential represents the different parts of a git credential URL
+// See also https://git-scm.com/docs/git-credential
+type GitCredential struct {
+	Protocol string
+	Host     string
+	Path     string
+	Username string
+	Password string
+}
+
+// CreateGitCredential creates a CreateGitCredential instance from a slice of strings where each element is a key/value pair
+// separated by '='.
+func CreateGitCredential(lines []string) (GitCredential, error) {
+	var credential GitCredential
+
+	if lines == nil {
```

Comment:
well, technically this will just mean that there won't be anything to extract. _nil_ on the other hand could panic.

Reply:
Instead of checking the lines length it would maybe be more useful to check whether there is at least a Protocol and Host at the end. I think for now we can live with how it is.